src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Remote/RemoteAutoInsertOptions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.Razor.Formatting;
+using Microsoft.CodeAnalysis.Razor.Settings;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.Razor.Remote;
+
+[DataContract]
+internal readonly record struct RemoteAutoInsertOptions
+{
+    [DataMember(Order = 0)]
+    public bool EnableAutoClosingTags { get; init; } = true;
+
+    [DataMember(Order = 1)]
+    public bool FormatOnType { get; init; } = true;
+
+    [DataMember(Order = 2)]
+    public RazorFormattingOptions FormattingOptions { get; init; } = new RazorFormattingOptions()
+    {
+        InsertSpaces = true,
+        TabSize = 4
+    };
+
+    public RemoteAutoInsertOptions()
+    {
+    }
+
+    public static RemoteAutoInsertOptions From(ClientSettings clientSettings, FormattingOptions formattingOptions)
+        => new()
+        {
+            EnableAutoClosingTags = clientSettings.AdvancedSettings.AutoClosingTags,
+            FormatOnType = clientSettings.AdvancedSettings.FormatOnType,
+            FormattingOptions = RazorFormattingOptions.From(formattingOptions, codeBlockBraceOnNextLine: false)
```

Comment:
This seems like a responsibility of the caller. The parameter should be a RazorFormattingOptions IMO, and if the caller can do nothing other than call From, then so be it. With this code as written, the caller might have a RazorFormattingOptions, have to convert to FormattingOptions to call this API, only for this to convert back and lose some data along the way.